modules/op-chains-generating-cosign-secret.adoc

```diff
@@ -0,0 +1,70 @@
+// This module is included in the following assemblies:
+// * secure/using-tekton-chains-for-openshift-pipelines-supply-chain-security.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="chains-generating-cosign-secret_{context}"]
+= Generating the cosign key pair by using the TektonConfig CR
+
+To use the `cosign` signing scheme for {tekton-chains} secrets, you can generate a `cosign` key pair that uses ECDSA encryption by setting the `generateSigningSecret` field in the `TektonConfig` custom resource (CR) to `true`.
+
+.Prerequisites
+
+* You installed the {oc-first} utility.
+* You are logged in to your {OCP} cluster with administrative rights for the `openshift-pipelines` namespace.
+
+.Procedure
+
+. Edit the `TektonConfig` CR by running the following command:
++
+[source,terminal]
+----
+$ oc edit TektonConfig config
+----
+
+. In the `TektonConfig` CR, set the `generateSigningSecret` value to `true`:
++
+.Example of creating an ECDSA cosign key pair by using the TektonConfig CR
+[source,yaml]
+----
+apiVersion: operator.tekton.dev/v1
+kind: TektonConfig
+metadata:
+  name: config
+spec:
+# ...
+  chain:
+    disabled: false
+    generateSigningSecret: true # <1>
+# ...
+----
+<1> The default value is `false`. Setting the value to `true` generates the `ecdsa` key pair.
+
+. After a few minutes, extract the public key from the secret and store it, so that you can use it to verify artifact attestations. Run the following command to extract the key:
++
+[source,terminal]
+----
+$ oc extract -n openshift-pipelines secret/signing-secrets --keys=cosign.pub
+----
+
+.Result
+
+The {pipelines-shortname} Operator generates an `ecdsa` type `cosign` key pair and stores it in the `signing-secrets` secret in the `openshift-pipelines` namespace. The secret includes the following files:
+
+* `cosign.key`: The private key
+* `cosign.password`: The password for decrypting the private key
+* `cosign.pub` The public key
+
+If a `signing-secrets` secret already exists, the Operator does not overwrite the secret.
+
+The `cosign.pub` file in your current directory contains the public key that is extracted from the secret.
+
+[WARNING]
+====
+If you set the `generateSigningSecret` field from `true` to `false`, the {pipelines-title} Operator overrides and empties any value in the `signing-secrets` secret.
+====
+
+The {pipelines-title} Operator does not provide the following functions to limit potential security issues:
```

Comment:
```suggestion
To limit potential security issues, the {pipelines-title} Operator does not provide the following functions:
```
Minor revision to emphasize the "why" of this statement.

Reply:
Good question, I'll raise this with the dev team (the formula was kept from the previous version).

Reply:
It was the other meaning, and it's fixed now.